src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexProvider.cs

```diff
@@ -2,27 +2,18 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.WebJobs.Host.Bindings;
 using Microsoft.Azure.WebJobs.Host.Executors;
-using Microsoft.Azure.WebJobs.Host.Storage;
+using Microsoft.Azure.WebJobs.Host.Timers;
 using Microsoft.Azure.WebJobs.Host.Triggers;
 
 namespace Microsoft.Azure.WebJobs.Host.Indexers
 {
     internal class FunctionIndexProvider : IFunctionIndexProvider
     {
-        private readonly ITypeLocator _typeLocator;
-        private readonly ITriggerBindingProvider _triggerBindingProvider;
-        private readonly IBindingProvider _bindingProvider;
-        private readonly IJobActivator _activator;
-        private readonly IFunctionExecutor _executor;
-        private readonly IExtensionRegistry _extensions;
-        private readonly SingletonManager _singletonManager;
-        private readonly TraceWriter _trace;
-
+        private Func<CancellationToken, Task<IFunctionIndex>> _initializeIndex;
```

Comment:
Agree that the generated code is 'worse', but it's better in the sense that we don't need to write it, don't need to add boilerplate to pass through another parameter, etc. 

The root of the problem/my frustration is the large number of parameters that are passed throughout the sdk - perhaps moving to C# 7 we could group some of these in tuples and use destructuring in parameters.